client/core/core.go

```diff
@@ -791,7 +791,7 @@ func (dc *dexConnection) compareServerMatches(srvMatches map[order.OrderID]*serv
 	dc.tradeMtx.RLock()
 	defer dc.tradeMtx.RUnlock()
 	for oid, trade := range dc.trades {
-		activeMatches := trade.activeMatches(false)
```

Comment:
So this is the one and only spot where we said `false` to not list matches that have at least reached a redeemed status, whether or not we hit `order.MatchConfirmed`, right?  I believe we did this here in `compareServerMatches` where we are trying to `"Identify active matches that are missing from server's response"` because we do not expect the server to report matches that have been redeemed. (server doesn't know or care about the `MatchConfirmed` status.

Lets look at what happens with these "missing" matches and if we can or should make an attempt to exclude (just here) matches that are at least redeemed by us in this use case.

Reply:
Does this create the wanted behavior https://github.com/decred/dcrdex/compare/d7224a25890c5e37ebccf8cd14d05219e7392f16..0b32eda28689ae01f458e9bdbc3faef404f32e44

We don't expect the server to have matches waiting for redeem confirms. I guess we also do not expect revoked matches? Server revoked matches for sure, but can we assume self revoked matches are also not being tracked by the server anymore? Do we care what the server says about those still?

Reply:
> Does this create the wanted behavior https://github.com/decred/dcrdex/compare/d7224a25890c5e37ebccf8cd14d05219e7392f16..0b32eda28689ae01f458e9bdbc3faef404f32e44

I think that diff is almost good, but the following line, `found` will be false:
https://github.com/JoeGruffins/dcrdex/blob/0b32eda28689ae01f458e9bdbc3faef404f32e44/client/core/core.go#L810

If you have a function which filters out the unnecessary matches from `activeMatches` before the `if len(activeMatches) == 0` check, it will work.

Reply:
Ok. How about this one https://github.com/decred/dcrdex/compare/0b32eda28689ae01f458e9bdbc3faef404f32e44..9a979de88a0769006305e65985fa370a3ce3cf96

Reply:
Yes this is what I meant, sorry I wasn't clear enough.